### DISCUSSION TITLE
How to deploy with AWS Lambda Web Adapter and monorepo?

### DISCUSSION THREAD

**Discussion by tomaszferens** [remix-run/remix#9434] [General]
I found this project https://github.com/awslabs/aws-lambda-web-adapter that let's you deploy Node.js servers to Lambda easily.

I'm having a hard time with deploying Remix with it for some reason.

I'm using pnpm workspaces and here is how my `apps/web/Dockerfile` looks like:

```docker
# apps/web/Dockerfile
FROM public.ecr.aws/docker/library/node:20-slim
COPY --from=public.ecr.aws/awsguru/aws-lambda-adapter:0.8.3 /lambda-adapter /opt/extensions/lambda-adapter

WORKDIR /var/task

ENV PORT=3000

RUN npm install -g pnpm

COPY . /var/task

RUN pnpm install
RUN pnpm --filter web build

RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 --home /home/remix remix

USER remix

EXPOSE 3000

CMD npm run start-web
```

The Dockerfile is pretty straightforward, it does pnpm install in root directory and runs a build command for the "web" app. The CMD is a script from `<root>/package.json`:

```
"start-web": "remix-serve ./apps/web/build/server/index.js"
```

Unfortunately it doesn't work. I see on CloudWatch very random errors like these:

```
Error: Cannot find package '/var/task/node_modules/.pnpm/@hookform+resolvers@3.3.4_react-hook-form@7.51.1/node_modules/react-hook-form/package.json' imported from /var/task/node_modules/.pnpm/@hookform+resolvers@3.3.4_react-hook-form@7.51.1/node_modules/@hookform/resolvers/zod/dist/zod.mjs
```
or
```
Error: Cannot find package '/var/task/node_modules/.pnpm/@tanstack+react-table@8.15.0_react-dom@18.2.0_react@18.2.0__react@18.2.0/node_modules/@tanstack/table-core/package.json' imported from /var/task/node_modules/.pnpm/@tanstack+react-table@8.15.0_react-dom@18.2.0_react@18.2.0__react@18.2.0/node_modules/@tanstack/react-table/build/lib/index.mjs
```
or
```
Error: EMFILE: too many open files, open '/var/task/node_modules/.pnpm/p-timeout@6.1.2/node_modules/p-timeout/index.js'
```

Looks like it can't find some random packages 🤔 

<img width="1624" alt="image" src="https://github.com/remix-run/remix/assets/17815179/fb3eefee-d603-45ad-a7f5-6432610215ce">

Anyone had a success with deploying Remix to Lambda with the Web Adapter?

**Comment by jenssogaard** [comment]
I'm also trying to run Remix on Lambda using AWS-lambda-adapter. Late to the party, but port 3000 is reserved. Not that it will solve your problem, but for others who might waste time on port issues:

* *AWS_LWA_PORT / PORT - Lambda Web Adapter will send traffic to this port. This is the port your web application listening on. Inside Lambda execution environment, the web application runs as a non-root user, and not allowed to listen on ports lower than 1024. Please also avoid port 9001 and 3000. Lambda Runtime API is on port 9001. CloudWatch Lambda Insight extension uses port 3000.* *
